core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -121,6 +122,13 @@ class ReplicaFetcherThread(name: String,
     }
   }
 
+  def warnIfInvalid(messageSet: ByteBufferMessageSet): Unit = {
+    if (messageSet.sizeInBytes > 0 && messageSet.validBytes <= 0)
+      error("Replication is failing due to a message that is greater than replica.fetch.max.bytes. This " +
+        "generally occurs when the max.message.bytes has been overridden to exceed this " +
+        "value and a suitably large message has also been sent.")
```

Comment:
Could we include the action that the user should take (e.g. increase replica.fetch.max.bytes in broker config)?